### PR TITLE
Games: Stable player assignment

### PR DESCRIPTION
**Pull request by garbear**
## Description

In https://github.com/xbmc/xbmc/pull/22036, I pulled in some work from my Player Manager, but unintentionally pulled in an unfinished change that causes a regression in v20.

The change in behavior has to do with how players are sorted into controller ports. In v19, the most recently active controller got port 1, etc. However, in the refactor, new, unfinished behavior was introduced, and the behavior contained [an error](https://github.com/xbmc/xbmc/pull/22036/files#diff-c9a9a33df7eabea92e79b782e14daef8bfd963d02a084906fc832a8ae6566d33R309) that caused joysticks to never be sorted; they're simply given a port depending on when they were detected by the OS.

As a result, on some Android boxes, "virtual" joysticks were now sorted in front of legit joysticks, causing input to be dropped.

This PR restores the old behavior of sorting ports by last activated joystick. It's a stop gap measure to solve the phantom joystick problem until the Player Manager is ready to allow control over port mappings.

I also added new logging when the port mapping changes to help debug future issues.

## Motivation and context

Reported several places, most recently here: https://forum.kodi.tv/showthread.php?tid=373412

## How has this been tested?

Test builds available here: https://github.com/garbear/xbmc/releases

Example logging with 5 controllers connected and a Multitap in port 2:

```
debug <general>: ===== Peripheral Map =====
debug <general>: /1/game.controller.snes.mouse:
debug <general>:     peripheral.joystick/0 [Xbox 360-compatible controller]
debug <general>:
debug <general>: /2/game.controller.snes.multitap/1/game.controller.snes:
debug <general>:     peripheral.joystick/1 [Xbox 360-compatible controller]
debug <general>:
debug <general>: /2/game.controller.snes.multitap/2/game.controller.snes:
debug <general>:     peripheral.joystick/2 [Xbox 360-compatible controller]
debug <general>:
debug <general>: /2/game.controller.snes.multitap/3/game.controller.snes:
debug <general>:     peripheral.joystick/3 [Xbox 360-compatible controller]
debug <general>:
debug <general>: /2/game.controller.snes.multitap/4/game.controller.snes:
debug <general>:     peripheral.joystick/4 [Wireless Controller]
debug <general>: ==========================
```

Example logging when the Multitap in port 2 is changed to a SNES controller:

```
debug <general>: ===== Peripheral Map =====
debug <general>: /1/game.controller.snes:
debug <general>:     peripheral.joystick/0 [Xbox 360-compatible controller]
debug <general>: 
debug <general>: /2/game.controller.snes:
debug <general>:     peripheral.joystick/1 [Xbox 360-compatible controller]
debug <general>: 
debug <general>: Disconnected:
debug <general>:     peripheral.joystick/2 [Xbox 360-compatible controller]
debug <general>:     peripheral.joystick/3 [Xbox 360-compatible controller]
debug <general>:     peripheral.joystick/4 [Wireless Controller]
debug <general>: ==========================
```

I also created a stripped-down version of the Player manager that I call "View Players" that simply allows you to view player mappings without changing anything:

screenshot00012

The View Players window is included in my test builds, but not in this PR.

## What is the effect on users?

* Fixed game controllers not working when "virtual" joysticks are detected

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

**Comment by garbear** [comment]
Reported fixed: https://forum.kodi.tv/showthread.php?tid=373412&pid=3158470#pid3158470

File: xbmc/games/addons/input/GameClientInput.cpp
Comment (Rechi) [inline xbmc/games/addons/input/GameClientInput.cpp:474]: GCC: `xbmc/games/addons/input/GameClientInput.cpp:474:36: warning: redundant move in initialization [-Wredundant-move]`
clang-tidy: `xbmc/games/addons/input/GameClientInput.cpp:474:27: warning: std::move of the const variable 'keyboard' has no effect; remove std::move() or make the variable non-const [performance-move-const-arg]`

File: xbmc/games/addons/input/GameClientInput.cpp
Comment (Rechi) [inline xbmc/games/addons/input/GameClientInput.cpp:543]: GCC: `xbmc/games/addons/input/GameClientInput.cpp:543:33: warning: redundant move in initialization [-Wredundant-move]`
clang-tidy: `xbmc/games/addons/input/GameClientInput.cpp:543:24: warning: std::move of the const variable 'mouse' has no effect; remove std::move() or make the variable non-const [performance-move-const-arg]`

**Review by Rechi — DISMISSED** [review]

**Comment by garbear** [comment]
I've pushed a new commit. It accomplishes a long-time goal of stable player sorting. When ports or controllers change around, players maintain their port. Even if multitaps are swapped out, when they're swapped back in, the port is remembered.

I'll wait for a confirmation that Android joysticks still work with the stable player sorting.

**Comment by garbear** [comment]
Android joysticks [confirmed](https://forum.kodi.tv/showthread.php?tid=373412&pid=3159135#pid3159135) working, will merge when I'm ready.

**Pull request merged**
